streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -447,6 +459,49 @@ private void removePartitions() {
         tasks.clear();
     }
 
+    public PartitionGrouper partitionGrouper() {
+        return partitionGrouper;
+    }
+
+    private void ensureCopartitioning(Collection<Set<String>> copartitionGroups) {
+        for (Set<String> copartitionGroup : copartitionGroups) {
+            ensureCopartitioning(copartitionGroup);
+        }
+    }
+
+    private void ensureCopartitioning(Set<String> copartitionGroup) {
+        int numPartitions = -1;
+
+        for (String topic : copartitionGroup) {
```

Comment:
Right. The current code supports only 1-to-1 copartitioning. We may want to generalize it further.
Anyway, your example is not possible with a current limitation of the consumer offset commit.